t/integration/test_canvas.py

```diff
@@ -232,6 +233,30 @@ def test_ready_with_exception(self):
         while not result.ready():
             pass
 
+
+    @flaky
+    def test_not_ready_with_http_exception(self):
+        g = group([add.s(1, 2), raise_custom_error.s()])
+        result = g.apply_async()
+        with pytest.raises(TypeError) as e:
+            result.join_native()
+
+    @flaky
+    def test_raises_wrong_exception(self):
+        g = group([add.s(1, 2), raise_custom_error.s()])
+        result = g.apply_async()
+        with pytest.raises(urllib.error.HTTPError) as e:
```

Comment:
Just so I'll get this right, this **should** fail with a HTTPError right?
Instead, it raises another exception?

Testcases should fail the build if there's a bug.

Reply:
Yes,you are correct.
It raises TypeError .Attached the report.html to get an idea.

![failed_test](https://user-images.githubusercontent.com/1801053/57770348-c46e6380-7718-11e9-9358-6d45b242ba1e.png)